script.js

```diff
@@ -0,0 +1,174 @@
+document.addEventListener('DOMContentLoaded', () => {
+
+  start_time();
+
+  let currentTab = getCurrentTabFromURL() || '#activity';
+
+  let triggerTabList = [].slice.call(document.querySelectorAll('#myTab a'));
+
+  triggerTabList.forEach(function (triggerEl) {
+    triggerEl.addEventListener('click', function (event) {
+      event.preventDefault();
+      const tabId = triggerEl.getAttribute('href');
+      showTab(tabId);
+      currentTab = tabId;
+      updateURL(currentTab);
+    })
+  });
+
+  window.addEventListener('popstate', function(event) {
+    currentTab = getCurrentTabFromURL() || '#activity';
+    showTab(currentTab);
+  });
+
+  HideElems();
+  AdaptHeader();
+
+  if (!document.location.hash) {
+    showTab('#activity');
+  }
+
+  function getCurrentTabFromURL() {
+    const hash = document.location.hash;
+    return hash ? hash : null;
+  }
+
+  function showTab(tabId) {
+    const activeTriggerEl = triggerTabList.find(triggerEl => triggerEl.getAttribute('href') === tabId);
+    if (activeTriggerEl) {
+      const tabTrigger = new bootstrap.Tab(activeTriggerEl);
+      tabTrigger.show();
+    }
+  }
+
+  function updateURL(tabId) {
+    history.pushState(null, null, tabId);
+  }
+});
+
+ymaps.ready(init);
+
+var myMap
+function init(){
+  myMap = new ymaps.Map("map_img", {
+    center: [51.83, 107.6],
+    zoom: 8
+  });
+
+  var loader = document.getElementById('loader');
+  loader.style.display = 'none';
+
+  myMap.events.add('tilesloadstart', function () {
+    loader.style.display = 'block';
+  });
+
+  myMap.events.add('tilesloaded', function () {
+    loader.style.display = 'none';
+  });
+
+  var myPlacemark = new ymaps.Placemark([51.83, 107.6], {
+    hintContent: 'Улан-Удэ'
+  });
+  myMap.geoObjects.add(myPlacemark);
+}
+
+function reload_map() {
+  myMap.destroy()
+  init()
+}
+
+function AdaptHeader() {
+  let menuMainHeader = document.querySelector("#menu_main_header .dropdown-menu");

```

Comment:
```suggestion
  const menuMainHeader = document.querySelector("#menu_main_header .dropdown-menu");

```
старайся сначала писать const а не let, если нужно будешь поменяешь, если не нужно будет, потом задумаешься стоит ли менять эту переменную вообще)